stan/math/prim/mat/fun/gp_exponential_cov.hpp

```diff
@@ -0,0 +1,253 @@
+#ifndef STAN_MATH_PRIM_MAT_FUN_GP_EXPONENTIAL_COV_HPP
+#define STAN_MATH_PRIM_MAT_FUN_GP_EXPONENTIAL_COV_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/scal/fun/square.hpp>
+#include <stan/math/prim/scal/fun/squared_distance.hpp>
+#include <stan/math/prim/scal/meta/return_type.hpp>
+#include <cmath>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/** Returns a Matern exponential covariance matrix with one input vector
+ *
+ * \f[ k(x, x') = \sigma^2 exp(-\frac{d(x, x')}{l}) \f]
+ *
+ * where \f$ d(x, x') \f$ is the squared distance, or the dot product.
+ *
+ * See Rausmussen & Williams et al 2006 Chapter 4.
```

Comment:
I'm looking at equation (4.18) in RW and not seeing the `sigma` term in there. Did you include this for convenience?

@avehtari, would you mind commenting on two things:
1. is this the right name for this function?
2. is this an appropriate parameterization for inclusion in Stan?

Reply:
@syclik see Betancourt's post on discourse: http://discourse.mc-stan.org/t/adding-gaussian-process-covariance-functions/237/42?u=drezap

I am following the conventions there.

Reply:
Thanks for pointing out the discussion. One thing that it's missing is the `jitter` term (also referred to as `nugget`).

I'm going to kick this question over to @avehtari. Specifically, Aki, what's the most common form of this kernel? 

@drezap, if we stick to this, I'd just like to see the note indicate that we're not using R&W's parameterization as-is.